packages/form-js-viewer/src/Form.js

```diff
@@ -104,12 +100,31 @@ export default class Form {
    * @param {Data} [data]
    */
   importSchema(schema, data = {}) {
-    this.importedData = clone(data);
-
-    this._setState({
-      data: clone(data),
-      errors: {},
-      schema: importSchema(schema).schema
+    this._importedData = null;
+    this._importedSchema = null;
+
+    return new Promise((resolve, reject) => {
+      const importer = this.get('importer');
+
+      schema = clone(schema);

```

Comment:
Actually, we only need to clone data twice. First, we clone it to avoid mutating the `data` that was passed and then we clone it again to keep a copy of `data` for resetting the form. I cleaned it up (cf. https://github.com/bpmn-io/form-js/pull/83/files#diff-53d9b079c7db646b79736d7ab07cd7d3108049b780d14637f41c188a4f6ffc03R105).